fpdf/fpdf.py

```diff
@@ -2029,28 +2048,38 @@ def cell(
                 "ignored"
             )
             border = 1
+        newpos_x = X.RIGHT
+        newpos_y = Y.TOP
+        if ln == 1:
+            newpos_x = X.LMARGIN
+            newpos_y = Y.NEXT
+        elif ln == 2:
+            newpos_x = X.LEFT
+            newpos_y = Y.NEXT
         # Font styles preloading must be performed before any call to FPDF.get_string_width:
         txt = self.normalize_text(txt)
         styled_txt_frags = self._preload_font_styles(txt, markdown)
         return self._render_styled_cell_text(
+            TextLine(styled_txt_frags, 0.0, 0, False),
             w,
             h,
-            styled_txt_frags,
             border,
-            ln,
-            align,
-            fill,
-            link,
-            center,
+            newpos_x=newpos_x,
+            newpos_y=newpos_y,
+            align=align,
+            fill=fill,
+            link=link,
+            center=center,
         )
 
     def _render_styled_cell_text(
         self,
+        text_line,
```

Comment:
The parameter class (`TextLine`) could be added as annotation here, to improve readability

Reply:
What do you mean by "annotation"?
In a docstring?

Reply:
I meant [type hints](https://docs.python.org/3/library/typing.html), like this:
```python
    def _render_styled_cell_text(
        self,
        text_line : TextLine,
```

Reply:
Ah, I've never done that before...
I'll figure it out!